libtransmission/tr-utp.cc

```diff
@@ -208,16 +208,12 @@ void tr_utpInit(tr_session* session)
 #endif
 
     session->utp_context = ctx;
+    session->utp_timer = session->timerMaker().create(timer_callback, session);
+    restart_timer(session);
 }
 
 bool tr_utpPacket(unsigned char const* buf, size_t buflen, struct sockaddr const* from, socklen_t fromlen, tr_session* ss)
 {
-    if (!ss->isClosing() && !ss->utp_timer)
-    {
-        ss->utp_timer = ss->timerMaker().create(timer_callback, ss);
-        reset_timer(ss);
-    }
-
```

Comment:
Setting up the periodic timer has been moved into `tr_utpInit()` so that it's always running. We need it to always be checking for timeouts etc. Previously it only got started once we'd received a uTP packet, which led to a checken-and-egg problem for the edge case of the Transmission session starting up with a torrent that only had TCP-only peers.